ballerina/agent-utils.bal

```diff
@@ -206,11 +219,22 @@ class Executor {
                     observation: observation.toString()
                 };
 
+                log:printError("Tool execution resulted in error",
```

Comment:
In these cases the agent might retry the operation to recover. In this case do we need to log an error? Wouldn't be adding a debug log is sufficient enough?

Reply:
I updated to debug logs